src/cpp/session/prefs/UserPrefsLayer.cpp

```diff
@@ -109,6 +109,19 @@ Error UserPrefsLayer::writePrefs(const core::json::Object &prefs)
       lastSync_ = prefsFile_.getLastWriteTime();
    }
 
+   // Modify the error to be more descriptive
+   if (isFileNotFoundError(error) && prefsFile_.getParent().exists())
+   {
+      error = Error(
+         error.getName(),
+         error.getCode(),
+         "Unable to save preferences. Please verify that " + 
+            prefsFile_.getParent().getAbsolutePath() +
+            " exists and it and all its children are owned by the user '" +
```

Comment:
For a non-technical audience, how about "the folder X and all the files inside it"?

Reply:
Oh, actually I realized I already changed this to `"Unable to save preferences. Please verify that " + prefsFile_.getAbsolutePath() + " exists and is owned by the user '" + system::username() + "'."`. I must not have pushed the change when I thought I did.